compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1412,6 +1412,10 @@ rustc_queries! {
         separate_provide_extern
     }
 
+    query is_general_impl(def_id: DefId) -> bool {
```

Comment:
I wonder if this needs to use a qualifier like "fully" or "maximally" to apply to "general" (I suppose "general" is better than "blanket"? at least a doc comment here could explain the connection).